File: Leetcode-Problems/Transformed Array.cpp
class Solution
{
public:
    vector<int> constructTransformedArray(vector<int> &nums)
    {
        int n = nums.size();
        vector<int> ans(n);
        for (int i = 0; i < nums.size(); i++)
        {
            int steps = nums[i];
            if (steps == 0)
            {
                ans[i] = nums[i];
            }
            else
            {
                steps = (i + steps) % n;
                if (steps < 0)
                    steps += n;
                ans[i] = nums[steps];
            }
        }
        return ans;
    }
};

File: Leetcode-Problems/Transformed Array.java
class Solution {
    public int[] constructTransformedArray(int[] nums) {
        int[] ans = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            int steps = nums[i];
            if (steps == 0) {
                ans[i] = nums[i];
            } else {
                steps = (i + steps) % nums.length;
                if (steps < 0) {
                    steps += nums.length;
                }
                ans[i] = nums[steps];
            }
        }
        return ans;
    }
}